Sage/carbon
# feat(dialogfullscreen): allow content in headers

### Proposed behaviour
Add left and right adaptive paddings to DialogFullScreen.

Add new `headerChildren` prop to allow the user to pass in components be be rendered to the right of the heading in `DialogFullScreen`. This content will wrap underneath the heading when there isn't enough room on the screen.

![image](https://user-images.githubusercontent.com/14963680/96234669-a5b96f80-0f91-11eb-8fd5-c0661f18e021.png)


### Current behaviour
Above functionality does not exist.

### Checklist


### Additional context

### Testing instructions
Responsive paddings: Look at the default story on DialogFullScreen and resize the window.

Header children: see the "with header children" story in storybook
